### PR TITLE
Dataman remove duplicate header files

### PR DESCRIPTION
**Pull request by xdwgood**
@bkueng   Can we remove these?  :smiley:

### CODE REVIEW

File: src/modules/dataman/dataman.cpp [hunk 1]
```diff
@@ -41,23 +41,9 @@
...
 #include <px4_posix.h>
-#include <px4_tasks.h>
-#include <stdio.h>
-#include <stdbool.h>
-#include <stdlib.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <systemlib/err.h>
-#include <queue.h>
-#include <string.h>
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: This is needed for `memset` and string methods

File: src/modules/dataman/dataman.cpp [hunk 2]
```diff
@@ -41,23 +41,9 @@
...
-#include <px4_config.h>
-#include <px4_defines.h>
 #include <px4_module.h>
 #include <px4_posix.h>
-#include <px4_tasks.h>
-#include <stdio.h>
-#include <stdbool.h>
-#include <stdlib.h>
-#include <errno.h>
-#include <fcntl.h>
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: This is needed for `open`

File: src/modules/dataman/dataman.cpp [hunk 3]
```diff
@@ -41,23 +41,9 @@
...
-#include <systemlib/err.h>
-#include <queue.h>
-#include <string.h>
-#include <semaphore.h>
-#include <unistd.h>
 #include <platforms/px4_getopt.h>
-#include <drivers/drv_hrt.h>
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: This is needed for the time

File: src/modules/dataman/dataman.cpp [hunk 4]
```diff
@@ -41,23 +41,9 @@
...
-#include <stdio.h>
-#include <stdbool.h>
-#include <stdlib.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <systemlib/err.h>
-#include <queue.h>
-#include <string.h>
-#include <semaphore.h>
-#include <unistd.h>
```
Comment (bkueng) [inline src/modules/dataman/dataman.cpp]: This is needed for `lseek`

File: src/modules/dataman/dataman.cpp [hunk 5]
```diff
@@ -41,23 +41,9 @@
  * @author David Sidrane
  */
 
-#include <px4_config.h>
```
Comment (dagar) [inline src/modules/dataman/dataman.cpp]: px4_config.h is needed for the FLASH_BASED_DATAMAN define on aerofc-v1.

**Comment by xdwgood** [comment]
> It's generally not good to rely on some headers indirectly including others.

@bkueng thanks for your all comments !  👍

I removed this because I found that `#include <px4_posix.h>` included a lot of header files.  So， Sould I close this extra PR？  😃

**Comment by bkueng** [comment]
Ok, I'm closing this, but smaller code cleanups are generally very much welcome, and also a good way to get to know PX4 in more depth.

**Pull request closed without merging**